src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -414,10 +418,20 @@ object CheckTypes extends Pass {
           errors append new InvalidConnect(info, mname, sx.loc.serialize, sx.expr.serialize)
         case sx: PartialConnect if !bulk_equals(sx.loc.tpe, sx.expr.tpe, Default, Default) =>
           errors append new InvalidConnect(info, mname, sx.loc.serialize, sx.expr.serialize)
-        case sx: DefRegister => sx.tpe match {
-          case AnalogType(w) => errors append new IllegalAnalogDeclaration(info, mname, sx.name)
-          case t if wt(sx.tpe) != wt(sx.init.tpe) => errors append new InvalidRegInit(info, mname)
-          case t =>
+        case sx: DefRegister => {
+          sx.tpe match {
+            case AnalogType(w) => errors append new IllegalAnalogDeclaration(info, mname, sx.name)
+            case t if wt(sx.tpe) != wt(sx.init.tpe) => errors append new InvalidRegInit(info, mname)
+            case t =>
+          }
+          sx.reset.tpe match {
+            case UIntType(IntWidth(w)) if (w == 1) =>
```

Comment:
I think leaving the cases separate is good for clarity. The fact that the case for UnknownWidth results in no error is an artifact of the pass ordering; I think it would be better to leave it separate (with a comment explaining that the check is imperfect) than to just go for maximum brevity.